data_structures/hashing/hash_table.ts

```diff
@@ -0,0 +1,212 @@
+/**
+ * Represents a hash table.
+ * Time complexity:
+ * - Set, Get, Delete, Has: O(1) on average, O(n) in the worst case.
+ * - Clear: O(m) where m is the number of buckets.
+ * - Keys, Values, Entires: O(n + m).
+ *
+ * @template K The key type.
+ * @template V The value type.
+ * @param size The size of the hash table.
+ * @param buckets The buckets in which to store the key-value pairs.
+ */
+export class HashTable<K, V> {
+    private size: number;
+    private buckets: HashTableEntry<K, V>[][];
+
+    constructor() {
+        this.size = 0;
+        this.buckets = [];
+        for (let i = 0; i < 16; i++) {
+            this.buckets.push([]);
+        }
+    }
+
+    /**
+     * Gets the size.
+     *
+     * @returns The size.
+     */
+    getSize(): number {
+        return this.size;
+    }
+
+    /**
+     * Sets a key-value pair.
+     *
+     * @param key The key.
+     * @param value The value.
+     */
+    set(key: K, value: V): void {
+        const index = this.hash(key);
+        const bucket = this.buckets[index];
+
+        if (bucket.length === 0) {
+            bucket.push(new HashTableEntry(key, value));
+            this.size++;
+            return;
+        }
+
+        for (const entry of bucket) {
+            if (entry.key === key) {
+                entry.value = value;
+                return;
+            }
+        }
+
+        bucket.push(new HashTableEntry(key, value));
+        this.size++;
+    }
+
+    /**
+     * Gets a value.
+     *
+     * @param key The key to get the value for.
+     * @returns The value or null if the key does not exist.
+     */
+    get(key: K): V | null {
+        const index = this.hash(key);
+        const bucket = this.buckets[index];
+
+        for (const entry of bucket) {
+            if (entry.key === key) {
+                return entry.value;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Deletes a key-value pair.
+     *
+     * @param key The key whose key-value pair to delete.
+     */
+    delete(key: K): void {
+        const index = this.hash(key);
+        const bucket = this.buckets[index];
+
+        for (const entry of bucket) {
+            if (entry.key === key) {
+                bucket.splice(bucket.indexOf(entry), 1);
+                this.size--;
+                return;
+            }
+        }
+    }
+
+    /**
+     * Checks if a key exists.
+     *
+     * @param key The key.
+     * @returns Whether the key exists.
+     */
+    has(key: K): boolean {
+        const index = this.hash(key);
+        const bucket = this.buckets[index];
+
+        for (const entry of bucket) {
+            if (entry.key === key) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Clears the hash table.
+     */
+    clear(): void {
```

Comment:
You could just call this in the constructor. Currently you're duplicating the code.